WatchKitTimePicker/TimePickerDataSource.swift

```diff
@@ -117,9 +119,17 @@ public class TimePickerDataSource {
             minutesPicker?.setRelativeWidth(0.5, withAdjustment: 0)
         }
         
+        pickersAreSetUp = true
+
         // set the initial values of the pickers
         if let initiallySelectedDate = initiallySelectedDate {
-            let dateComponents = Calendar.current.dateComponents(Set(arrayLiteral: .hour, .minute), from: initiallySelectedDate)
+            setDate(withDate: initiallySelectedDate)
+        }
+    }
+
+    public func setDate(withDate date: Date) {
+        if pickersAreSetUp {
```

Comment:
I don't think the `pickersAreSetUp` property is necessary here. Instead, I think we should change `setup(withInitiallySelectedDate:) to just `setup()` and add a call to `setup()` from `init`.